rust/tool/tests/common/mod.rs

```diff
@@ -80,14 +101,16 @@ pub fn setup_generation_link_from_toplevel(
 ///
 /// Accepts the temporary directory as a parameter so that the invoking function retains control of
 /// it (and when it goes out of scope).
-pub fn setup_toplevel(tmpdir: &Path) -> Result<PathBuf> {
+pub fn setup_toplevel(tmpdir: &Path, system_double: &str) -> Result<PathBuf> {
```

Comment:
The way you implemented this, I think, e don't really need the parameter here. We can just call `target_system_double()` from within `setup_toplevel`.

Reply:
I wanted to ensure that everyone had the same target system double if we decided to pass mixed systems

Reply:
I'd say we reduce the noise now (call `target_system_double()`  from within `setup_toplevel()`) and then we can still add it as a parameter when we actually need/use it.

Reply:
Agreed!